### PR TITLE
Adds yaml_to_json_converter.py

### PR DESCRIPTION
**Pull request by abhi12ravi**
Steps performed by the method:

1. It takes two optional parameters: input_folder (default: "models") and output_folder (default: "models-json").
2. It checks if the input folder exists and creates the output folder if it doesn't exist.
3. It iterates through all files in the input folder with '.yaml' or '.yml' extensions.
4. For each YAML file, it reads the content, converts it to JSON, and writes it to a corresponding JSON file in the output folder.
5. It handles various exceptions that might occur during the process, such as file not found, permission errors, YAML parsing errors, and JSON conversion errors.
6. It returns a list of successfully converted file names.

### CODE REVIEW

**Comment by abhi12ravi** [comment]
Fix for issue: https://github.com/OpenEthicsAI/OEDP/issues/1

File: API/models-json/Open-Ethics-Data-Passport.json [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  "description": "OEDP Model",
+  "properties": {
+    "id": {
+      "type": "string",
+      "format": "uuid"
+    },
+    "datasets": {
+      "type": "array",
+      "items": {
+        "$ref": "./Dataset.yaml"
```
Comment (nikiluk) [inline API/models-json/Open-Ethics-Data-Passport.json:13]: @abhi12ravi the reference should point to the corresponding JSON file but it seems that it points to the YAML. I suggest  not to spend much time on the python fixing but directly to fix it in the JSON and their nested children JSON schemas

File: API/models-json/Open-Ethics-Data-Passport.json [hunk 2]
```diff
@@ -0,0 +1,46 @@
+{
+  "title": "Open Ethics Data Passport",
+  "type": "object",
+  "description": "OEDP Model",
+  "properties": {
+    "id": {
+      "type": "string",
+      "format": "uuid"
+    },
+    "datasets": {
```
Comment (nikiluk) [inline API/models-json/Open-Ethics-Data-Passport.json:10]: @abhi12ravi it seems that the schema still follows the old structure, we'd need to advance it to the new one. E.g. a passport of the product may employ several models, which in turn may employ several datasets.

In the previous version we considered unimodular AI products but, as you know, today many products may embed several models simultaneously.

In addition, we may require to describe models alone when there's absolutely no info on the datasets. 

From the issue description
> Follow the newest hierarchy described here https://openethics.ai/oedp/ (not yet in the YAML) to incorporate the idea that there could be multiple models supporting a product

Comment (nikiluk) [reply API/models-json/Open-Ethics-Data-Passport.json:10]: e.g. we'd need to have a new branch element called "models" which will be an array of Models defined by a new schema called "Model.JSON" and a model schema in turn will contain datasets as optional

File: API/models-json/Open-Ethics-Data-Passport.json [hunk 3]
```diff
@@ -0,0 +1,46 @@
+{
+  "title": "Open Ethics Data Passport",
+  "type": "object",
+  "description": "OEDP Model",
```
Comment (nikiluk) [inline API/models-json/Open-Ethics-Data-Passport.json:4]: @OpenEthicsAI @abhi12ravi I'm noticing that my previous definitions are misleading as i referred to the data model in the structural sense and I don't want to confuse with the AI models which we want to describe with the schemas, so we'd change the naming to the data schema

File: API/models-json/Open-Ethics-Data-Passport.json [hunk 4]
```diff
@@ -0,0 +1,46 @@
+{
```
Comment (nikiluk) [inline API/models-json/Open-Ethics-Data-Passport.json:1]: @abhi12ravi the header of our every schema module/json file should contain required metadata

1) contain `$id` (we'll later upload it to openethics.ai for everyone to see) and will submit to the schemastore.org
2) contain `  "$schema": "https://json-schema.org/draft/2020-12/schema"`

as defined in the metaschema specs https://json-schema.org/draft/2020-12/release-notes

File: API/models-json/Open-Ethics-Data-Passport.json (hunk 1 above)
Comment (abhi12ravi) [reply API/models-json/Open-Ethics-Data-Passport.json]: @nikiluk Got it. Now, it points to `Dataset.json file`. Hope that's okay?

`        "$ref": "./Dataset.json"
`

File: API/models-json/Open-Ethics-Data-Passport.json (hunk 2 above)
Comment (abhi12ravi) [reply API/models-json/Open-Ethics-Data-Passport.json:10]: @nikiluk I'm unable to follow, could we have a call around this?

**Pull request merged**